Patch edk2-ovmf file to fix UEFI support

priteau opened the pull request:
The file 50-edk2-ovmf-cc.json is used by Nova in get_loader() when lauching UEFI instances without Secure Boot.

It only supports machine type pc-q35-rhel8.5.0. However, with recent nova-compute container images built on CentOS Stream 8, the machine type is pc-q35-rhel8.6.0, which results in failures to launch instances with:

    nova.exception.UEFINotSupported: UEFI is not supported

There is a fix for RHEL9/c9s [1], but not for RHEL8/c8s. Also see [2] for more details.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=2090752
[2] https://bugs.launchpad.net/nova/+bug/1955035

JohnGarbutt reviewed (dismissed):
That looks like a good quick patch.

priteau commented:
I've just realised this needs to be made specific to centos as it might fail on Ubuntu.

JohnGarbutt commented:
> I've just realised this needs to be made specific to centos as it might fail on Ubuntu

oops, you are quite correct.

priteau committed:
Patch edk2-ovmf file to fix UEFI support

The file 50-edk2-ovmf-cc.json is used by Nova in get_loader() when
lauching UEFI instances without Secure Boot.

It only supports machine type pc-q35-rhel8.5.0. However, with recent
nova-compute container images built on CentOS Stream 8, the machine type
is pc-q35-rhel8.6.0, which results in failures to launch instances with:

    nova.exception.UEFINotSupported: UEFI is not supported

There is a fix for RHEL9/c9s [1], but not for RHEL8/c8s. Also see [2]
for more details.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=2090752
[2] https://bugs.launchpad.net/nova/+bug/1955035
